app/pages/adduser/contextCreatorErrorHelper.js

```diff
@@ -0,0 +1,31 @@
+import { errorMessages } from './errorStrings';
+
+export const formatErrors = errors => errors.reduce((acc, error) => {
+  const errorMessageLocator = `${error.field[0].toLowerCase()}${error.field.slice(1)}`;
+  if (acc[errorMessageLocator]) acc[errorMessageLocator].push(errorMessages[error.id]);
+  else acc[errorMessageLocator] = [errorMessages[error.id]];
+  return acc;
+}, {});
```

Comment:
If I am understanding this function correctly. I think it's trying to find the error message from the errorStrings? I think the complexity of finding the errorMessage could be reduced by moving this into the manifest. The field value should match the questionId and then from there in the manifest we find the errorMessage to use using the id of the error.

manifest example;
    {
      "id": "firstName",
      "mainAdvice": "First name"
      "errorMessages": {
          FirstNameRequired: 'First name is required',
            FirstNameTooLong: 'First name is too long',
            FirstNameTooLong: 'First name is too long',
      },
    },

Reply:
The data is sent back with an upper case first letter and we want it lower case.
We still need the errors in a specific format to send to the component: `{[fieldName]: 'error message' }`

Reply:
it comes in like this : `[
  { field: 'emailAddress', id: 'EmailRequired' },
  { field: 'emailAddress', id: 'EmailTooLong' },
  { field: 'lastName', id: 'LastNameTooLong' },
]`

Reply:
also the questions in the manifest are in an array so you need to map over the messages then map over the questions inside that

Reply:
I think that's why I ended up using the id of the questions as a key in the manifest to make this lookup easier.